lib/puppet/provider/package/puppet_gem.rb

```diff
@@ -5,10 +5,7 @@
   has_feature :versionable, :install_options, :uninstall_options
 
   if Puppet::Util::Platform.windows?
-    # On windows, we put our ruby ahead of anything that already
-    # existed on the system PATH. This means that we do not need to
-    # sort out the absolute path.
-    commands :gemcmd => "gem"
+    commands :gemcmd => File.join(ENV['PUPPET_DIR'].to_s, 'bin', 'gem.bat')
```

Comment:
I believe `PUPPET_DIR` defaults to `C:\Program Files\Puppet Labs\Puppet\puppet`, but we've been bitten by the double puppet directories in `C:\...\Puppet\puppet`, so wanted to double check.

The environment variable I think is fine in general, but there may be an issue on non-US locales due to ruby not handling environment variables. You might want to use `Puppet::Util.get_env('PUPPET_DIR')` instead as it will correctly transcode from UTF-16LE to UTF-8. Another option is to use the `env_install_dir` fact that `environment.bat` sets?